UnityProject/Assets/Scripts/Objects/Engineering/PowerGenerator.cs

```diff
@@ -52,14 +52,21 @@ private enum SpriteState
 
 		void Awake()
 		{
-			registerTile = GetComponent<RegisterTile>();
-			securable = GetComponent<WrenchSecurable>();
-			baseSpriteHandler = GetComponentInChildren<SpriteHandler>();
-			electricalNodeControl = GetComponent<ElectricalNodeControl>();
+
+			Initialise();
 		}
 
 		public void OnSpawnServer(SpawnInfo info)
 		{
+			Initialise();
+		}
+
+		public void Initialise()
+		{
+			registerTile = GetComponent<RegisterTile>();
+			securable = GetComponent<WrenchSecurable>();
+			baseSpriteHandler = GetComponentInChildren<SpriteHandler>();
+			electricalNodeControl = GetComponent<ElectricalNodeControl>();
 			var itemStorage = GetComponent<ItemStorage>();
 			itemSlot = itemStorage.GetIndexedItemSlot(0);
 			securable.OnAnchoredChange.AddListener(OnSecuredChanged);
```

Comment:
you're adding two listeners to this now, it will be called twice

Reply:
I don't understand what's wrong with the way it is already, please explain further what you're trying to accomplish

Reply:
On spawn server isn't called when it's already on the scene I think

Reply:
its called at roundstart

Reply:
tho  there are errors on staging relating to it not be initialised properlybefore being used

Reply:
TryToggleOn  errored  because itemSlot was null

Reply:
Yeah, ItemSlot / ItemStorage stuff are snowflakes and can't be used in Awake. Start might be ok? Would have see how they work.

Reply:
Turns out, that same storage method `GetIndexedItemSlot` is used elsewhere in `Start()` so presumably it is ok there.